service/controller/machine_pool.go

```diff
@@ -159,8 +175,23 @@ func NewMachinePoolResourceSet(config MachinePoolConfig) ([]resource.Interface,
 		}
 	}
 
+	var cloudconfigblobResource resource.Interface
+	{
+		c := cloudconfigblob.Config{
+			ClientFactory: clientFactory,
+			CtrlClient:    config.K8sClient.CtrlClient(),
+			Logger:        config.Logger,
+		}
+
+		cloudconfigblobResource, err = cloudconfigblob.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
 	resources := []resource.Interface{
 		sparkResource,
+		cloudconfigblobResource,
```

Comment:
This new handler is executed right after the `sparkResource` tries to create the `Secret` with the right `cloudconfig`.